src/servers/src/server.rs

```diff
@@ -157,6 +158,15 @@ impl AcceptTask {
                 let addr = listener.local_addr()?;
                 info!("{name} server started at {addr}");
 
+                // set keep-alive
+                if keep_alive_secs > 0 {
+                    let socket_ref = socket2::SockRef::from(&listener);
```

Comment:
is it to set on listener or on socket returned from accept?

Reply:
It looks like this can be inherited by the client socket https://stackoverflow.com/questions/43441913/so-keepalive-set-on-the-server-socket-or-on-a-per-client-basis